functions-continued/variables-are-references.md

```diff
@@ -0,0 +1,648 @@
+`apples`, `oranges`, and `bananas` **_all_** had the same object ID, because their value was `26`. `26` is an integer, and **_integers are immutable data types_**.
+
+This subtle detail may seem more clear after comparing it to mutable data types.
+
+## Lists and Dictionaries Are Mutable
+
+Mutable objects, on the other hand, do _not_ change object IDs when the object is modified or _mutated_. Mutable objects _keep_ their current object IDs, even after being mutated.
+
+Examples of mutation and modification include:
+
+- Appending an item to a list
+- Removing an item from a list
+- Changing the value of an item in a list
+- Adding a key-value pair to a dictionary
+- Removing a key-value pair from a dictionary
+- Changing the value of a key or value in a dictionary
+
+Mutation and modification **_does not_** include reassignment.
+
+### Identical Mutable Objects Can Have Different IDs
+
+Read this example code, then read the story that goes along with it.
+
+```python
+mangoes = ["hearts", "stars", "horseshoes"]
+
+kiwis = ["hearts", "stars", "horseshoes"]
+```
+
+Imagine running this code. After the first time our Python interpreter sees `["hearts", "stars", "horseshoes"]`, it puts this list object into memory. The _second_ time the program sees `["hearts", "stars", "horseshoes"]`, _because lists are mutable_, it puts this list into _another, separate_ place in memory.
+
+**Even if these objects have an identical value**, we can prove that these two mutable objects are **in different places in memory** by using object IDs.
+
+```python
+mangoes = ["hearts", "stars", "horseshoes"]
+kiwis = ["hearts", "stars", "horseshoes"]
+
+print("id of mangoes:", id(mangoes))
+print("id of kiwis:", id(kiwis))
+```
+
+When we run the above code, we may see output like this:
+
+```
+id of mangoes: 4346624576
+id of kiwis: 4346599360
+```
+
+The variables `mangoes` and `kiwis` both have a value of `["hearts", "stars", "horseshoes"]`. However, **they have different object IDs** and occupy different places in memory.
+
+### Modifying a Mutable Object Doesn't Affect Its ID
+
+We can modify a mutable object, and the mutable object will continue to have the same object ID.
+
+```python
+mangoes = ["hearts", "stars", "horseshoes"]
+
+print("[Before modification]", "id of mangoes:", id(mangoes))
+
+mangoes.append("clovers")
+
+print("[After modification]", "id of mangoes:", id(mangoes))
+print("[After modification]", "value of mangoes", mangoes)
+```
+
+When we run the above code, we may see output like this:
+
+```
+[Before modification] id of mangoes: 4368230016
+[After modification] id of mangoes: 4368230016
+[After modification] value of mangoes ['hearts', 'stars', 'horseshoes', 'clovers']
+```
+
+_Even after modification_, the mutable object's object ID remains the same.
+
+## Multiple References to Mutable Objects
+
+As mentioned earlier, variables are references to objects in memory.
+
+Two different variables can end up referring to _the same_ object in memory.
```

Comment:
I discussed this situation with immutable objects as well, so now this tone sounds strange.  Add some transition text to recognize the similarity to the previous discussion.

```suggestion
As with immutable objects, we can end up with two different variables referring to _the same_ object in memory. But with mutable objects, the situation is a little more interesting.
```